### DISCUSSION TITLE
Additional fields for NativeFirebaseError.

### DISCUSSION THREAD

**Discussion by johnkmzhou** [invertase/react-native-firebase#4202] [Ideas]
It would be nice for NativeFirebaseError to adhere more closer to [firebase.auth.Error](https://firebase.google.com/docs/reference/js/firebase.auth.Error) by adding email and credential fields:

> Errors with the code auth/account-exists-with-different-credential will have the additional fields email and credential which are needed to provide a way to resolve these specific errors.

**Comment by mikehardy** [comment]
@johnkmzhou if you reach in to node_modules and add some print statements around where the error is generated, you can see if the data is actually available. I am not sure it is, there is no mention of it in the native Android doc: https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseAuthUserCollisionException or iOS https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes/FIRAuth#-createuserwithemail:password:completion:

They specifically mention in iOS:

> `FIRAuthErrorCodeEmailAlreadyInUse` - Indicates the email used to attempt sign up already exists. Call fetchProvidersForEmail to check which sign-in mechanisms the user used, and prompt the user to sign in with one of those.

and

> FIRAuthErrorCodeAccountExistsWithDifferentCredential - Indicates the email asserted by the credential (e.g. the email in a Facebook access token) is already in use by an existing account, that cannot be authenticated with this sign-in method. Call fetchProvidersForEmail for this user’s email and then prompt them to sign in with any of the sign-in providers returned. This error will only be thrown if the “One account per email address” setting is enabled in the Firebase console, under Auth settings.

...which is the strategy I have used with success

For native to add the information you want means native would have to call fetchProvidersForEmail itself - extra roundtrips on the network etc - is that what you mean?

**Comment by johnkmzhou** [comment]
Taking a look at the [nativeError](https://github.com/invertase/react-native-firebase/blob/master/packages/app/lib/internal/registry/nativeModule.js#L47) the fields aren't available. So it would be great for the the native code to provide it. Email from `getEmail` in Android's FirebaseAuthUserCollisionException and the NSError.userInfo dictionary object in iOS's [AuthErrorUserInfoEmailKey](https://firebase.google.com/docs/reference/swift/firebaseauth/api/reference/Constants#autherroruserinfoemailkey).

I don't need native to call fetchProvidersForEmail. I can do that from my own code.

**Comment by mikehardy** [comment]
PRs happily accepted :pray: - we rely on the community to move things like this forward though, in the absence of a PR, likely nothing will happen
